test/kubernetes_test.go

```diff
@@ -3,72 +3,169 @@
-	{"mynginx.*.svc.coredns.local.", 1, 1},                       // One A record, via wildcard namespace
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One A record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two A records, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two A records, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // Two A record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // Two A record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two A records, via namespace and service wildcard
-}
-
-// Test data for SRV records
-var testdataLookupSRV = []struct {
-	Query            string
-	TotalAnswerCount int
-	//	ARecordCount     int
-	SRVRecordCount int
-}{
-	// Matching queries
-	{"mynginx.demo.svc.coredns.local.", 1, 1}, // One SRV record, should exist
-
-	// Failure queries
-	{"mynginx.test.svc.coredns.local.", 0, 0},                     // One SRV record, is not exposed
-	{"someservicethatdoesnotexist.demo.svc.coredns.local.", 0, 0}, // Record does not exist
-
-	// Namespace wildcards
-	{"mynginx.*.svc.coredns.local.", 1, 1},                       // One SRV record, via wildcard namespace
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One SRV record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two (mynginx, webserver) SRV record, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two (mynginx, webserver) SRV record, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // One SRV record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // One SRV record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two SRV record, via namespace and service wildcard
-}
-
-func TestKubernetesIntegration(t *testing.T) {
-
-	// t.Skip("Skip Kubernetes Integration tests")
-	// subtests here (Go 1.7 feature).
-	testLookupA(t)
-	testLookupSRV(t)
+// Test data
+// TODO: Fix the actual RR values
+
+var dnsTestCases = []test.Case{
+	{
+		Qname: "mynginx.demo.svc.coredns.local.", Qtype: dns.TypeA,
+		Rcode: dns.RcodeSuccess,
+		Answer: []dns.RR{
+			test.A("mynginx.demo.svc.coredns.local.      1800    IN      A       10.3.0.10"),
```

Comment:
For future reference ('cause it is hard to find): a TTL of 303 means: don't check ttl